xds/src/main/java/com/linecorp/armeria/xds/AbstractResourceNode.java

```diff
@@ -24,7 +24,7 @@
 import io.envoyproxy.envoy.config.core.v3.ConfigSource;
 import io.grpc.Status;
 
-abstract class AbstractResourceNode<T> implements ResourceNode<AbstractResourceHolder> {
+abstract class AbstractResourceNode<T extends ResourceHolder<?>> implements ResourceNode<T> {
```

Comment:
How about?
```suggestion
abstract class AbstractResourceNode<T extends Message> implements ResourceNode<ResourceHolder<T>> {
```

Reply:
If so, we should cast in some cases because we cannot use the concrete type.
e.g.
Can't use `ClusterResourceHolder` when creating `ClusterSnapshot`. Instead we should use `ResourceHolder<Cluster>`. 
https://github.com/line/armeria/blob/012b9d14c9d6d6ef7004b60c4adbbf8652455e36/xds/src/main/java/com/linecorp/armeria/xds/ClusterResourceNode.java#L91-L99
Some of the holders have its own API:
https://github.com/line/armeria/blob/012b9d14c9d6d6ef7004b60c4adbbf8652455e36/xds/src/main/java/com/linecorp/armeria/xds/ConfigSourceClient.java#L70